controller/lib/hal/hal_stm32.cpp

```diff
@@ -0,0 +1,856 @@
+/* Copyright 2020, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#if defined(BARE_STM32)
+
+#include "hal.h"
+#include "stm32.h"
+
+// This is the main stack 
+uint32_t mainStack[ 500 ];
+
+// local data 
+static volatile uint32_t msCount;
```

Comment:
Good point (although I pity the poor sole stuck on any ventilator for that long).
You realize that the hal.h uses a 32-bit value for this, right?  That's really the only thing this is used for, returning the 32-bit result to millis().  That and a busy delay which hopefully wont be for more then a month.
I'll change this to a 64-bit value, but this also needs to be fixed in hal.h.  Want me to do that as part of this PR?

Reply:
> I'll change the definition of millis() in hal.h to return an unsigned 64-bit integer.
It looks like the Time class in units.h uses a 32-bit int for time in milliseconds, so that should be fixed as well.

Update your tree.  :)

Reply:
I'll rebase, should have done that first.

Reply:
I'm getting stuck trying to rebase my branch.  `git rebase master` doesn't seem to do anything.  The changes in master aren't getting applied to my branch.

I tried `git rebase --onto master `, but that just made a hash of everything.  Lots of lost changes on my branch.

I've been researching this for the last hour, but can't seem to figure out where I've gone wrong with this branch.  Any tips?

Reply:
Actually, I may have just fumbled my way through this.  
Seems like I had had to checkout master on my branch, then checkout my branch again, then rebase.  Not sure why that made a difference, but it now seems to have merged properly.
I'm still a bit lost with git, but I think everything worked out.